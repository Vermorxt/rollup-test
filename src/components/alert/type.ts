import { Ui_colors } from '../_types/colors/ui_colors'
import { HTMLProps, MouseEventHandler } from 'react'

export interface Ui_AlertProps extends HTMLProps<HTMLDivElement>, Ui_colors {
  noIcon?: boolean
  onDenyButtonText?: string
  onAcceptButtonText?: string
  onDeny?: MouseEventHandler<HTMLButtonElement> | undefined
  onAccept?: MouseEventHandler<HTMLButtonElement> | undefined
}
