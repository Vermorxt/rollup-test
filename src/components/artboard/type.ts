import { Ui_colors } from './../_types/colors/ui_colors'
import { HTMLProps } from 'react'

export interface Ui_ArtboardProps extends HTMLProps<HTMLSpanElement>, Ui_colors {
  phone1?: boolean
  phone2?: boolean
  phone3?: boolean
  phone4?: boolean
  phone5?: boolean
  phone6?: boolean
  horizontal?: boolean
  demo?: boolean
}
