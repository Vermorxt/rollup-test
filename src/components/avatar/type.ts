import { HTMLProps } from 'react'
import { Ui_Sizes } from '../_types/sizes/sizes'

interface Ui_Avatar_radius {
  roundedXl?: boolean
  rounded2xl?: boolean
  rounded3xl?: boolean
  roundedLg?: boolean
  roundedFull?: boolean
}

interface Ui_Avatar_ring {
  ring?: boolean
  ringPrimary?: boolean
  ringSecondary?: boolean
  ringOffset?: '0' | '1' | '2' | '4' | '8'
  ringOffsetBackground?: string // NOTE: https://tailwindcss.com/docs/ring-offset-color
}

interface Ui_Avatar_shapes {
  squircle?: boolean
  hexagon?: boolean
  triangle?: boolean
}

interface Ui_Avatar_online {
  online?: boolean
  offline?: boolean
}

export interface Ui_AvatarProps
  extends HTMLProps<HTMLDivElement>,
    Ui_Sizes,
    Ui_Avatar_radius,
    Ui_Avatar_shapes,
    Ui_Avatar_online,
    Ui_Avatar_ring {
  src?: string
  usePlaceHolder?: string // NOTE: content of last avatar item (e.g.: +99)
}
