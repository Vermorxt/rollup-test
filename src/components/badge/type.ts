import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'

export type TooltipPositionProps = 'top' | 'left' | 'bottom' | 'right'

export interface Ui_Badge_variants {
  outline?: boolean
  active?: boolean
  disabled?: boolean
  glass?: boolean
}

export interface Ui_Badge_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export interface Ui_BadgeProps
  extends HTMLProps<HTMLSpanElement>,
    Ui_colors,
    Ui_Sizes,
    Ui_Badge_variants,
    Ui_Badge_singleClass {
  wide?: boolean
  full?: boolean
}
