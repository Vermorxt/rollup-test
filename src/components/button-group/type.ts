import { HTMLProps } from 'react'

export interface Ui_ButtonGroupProps extends HTMLProps<HTMLDivElement> {}
