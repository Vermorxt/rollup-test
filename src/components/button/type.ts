import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
import { Ui_Sizes } from '../_types/sizes/sizes'
import { TooltipProps } from '../_types/tooltip/tooltip-props'

export interface Ui_Button_variants {
  outline?: boolean
  active?: boolean
  disabled?: boolean
  glass?: boolean
}

export interface Ui_Button_singleClass {
  loading?: boolean
  noAnimation?: boolean
}

export interface Ui_Button_sizes extends Ui_Sizes {
  wide?: boolean
  block?: boolean
  circle?: boolean
  square?: boolean
  full?: boolean
}

export type Ui_ButtonType = 'button' | 'submit' | 'reset' | undefined

export interface Ui_ButtonProps
  extends HTMLProps<HTMLButtonElement>,
    Ui_colors,
    Ui_Button_sizes,
    Ui_Button_variants,
    Ui_Button_singleClass {
  tooltip?: TooltipProps
  type?: Ui_ButtonType
}
