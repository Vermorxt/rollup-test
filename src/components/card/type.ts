import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'
interface Ui_Card_style {
  bgPrimary?: boolean
  bgSecondary?: boolean
  bgNeutral?: boolean
  textPrimaryContent?: boolean
  textSecondaryContent?: boolean
  textNeutralContent?: boolean
  bgBase?: '100' | '200' | '300'
}
export interface Ui_CardProps extends HTMLProps<HTMLDivElement>, Ui_Card_style {
  imageAsBackground?: boolean
  compact?: boolean
  itemsCenter?: boolean
  textCenter?: boolean
  glass?: boolean
  cardSide?: boolean
  cardSideLarge?: boolean
  w96?: boolean
  actionCenter?: boolean
  actionStart?: boolean
  actionSpaceBetween?: boolean
}
