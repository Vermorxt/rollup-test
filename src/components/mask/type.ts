import { HTMLProps } from 'react'
import { Ui_colors } from '../_types/colors/ui_colors'

export interface Ui_Mask_shapes {
  squircle?: boolean
  hexagon?: boolean
  hexagon2?: boolean
  triangle?: boolean
  triangle2?: boolean
  triangle3?: boolean
  triangle4?: boolean
  heart?: boolean
  decagon?: boolean
  pentagon?: boolean
  diamond?: boolean
  square?: boolean
  circle?: boolean
  parallelogram?: boolean
  parallelogram2?: boolean
  parallelogram3?: boolean
  parallelogram4?: boolean
  star?: boolean
  star2?: boolean
}

export interface Ui_MaskProps extends HTMLProps<HTMLImageElement>, Ui_colors, Ui_Mask_shapes {
  src?: string
}
