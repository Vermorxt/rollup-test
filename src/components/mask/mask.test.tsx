import '@testing-library/jest-dom'
import { FC } from 'react'
import { SpecialProps } from '../../_tests-utils/it-should-have-class'
import { SkipTests, testGenerator, TestGeneratorProps } from '../../_tests-utils/_utils/test-generator'
import { Ui_MaskProps } from './type'
import Ui_Mask from './mask'
import React from 'react'

const componentName = Ui_Mask
const selector = 'mask'
const refType = HTMLElement
const defaultProps: Ui_MaskProps = {}
const skipTests = [SkipTests.SkipChildren]

interface SpecialPropsMerged extends SpecialProps {
  className: string
  prop?: Ui_MaskProps
}

const component = (
  <Ui_Mask {...defaultProps} ref={null}>
    Test
  </Ui_Mask>
) as unknown as FC<{}>

const testName = `@cms/core/${componentName}`

const componentPropsColors: SpecialPropsMerged[] = []

const componentPropsDefault: SpecialPropsMerged[] = []

const componentPropsShapes: SpecialPropsMerged[] = [
  { prop: { squircle: true }, className: `${selector}-squircle` },
  { prop: { triangle: true }, className: `${selector}-triangle` },
  { prop: { triangle2: true }, className: `${selector}-triangle-2` },
  { prop: { triangle3: true }, className: `${selector}-triangle-3` },
  { prop: { triangle4: true }, className: `${selector}-triangle-4` },
  { prop: { heart: true }, className: `${selector}-heart` },
  { prop: { hexagon: true }, className: `${selector}-hexagon` },
  { prop: { hexagon2: true }, className: `${selector}-hexagon-2` },
  { prop: { decagon: true }, className: `${selector}-decagon` },
  { prop: { pentagon: true }, className: `${selector}-pentagon` },
  { prop: { diamond: true }, className: `${selector}-diamond` },
  { prop: { square: true }, className: `${selector}-square` },
  { prop: { circle: true }, className: `${selector}-circle` },
  { prop: { parallelogram: true }, className: `${selector}-parallelogram` },
  { prop: { parallelogram2: true }, className: `${selector}-parallelogram-2` },
  { prop: { parallelogram3: true }, className: `${selector}-parallelogram-3` },
  { prop: { parallelogram4: true }, className: `${selector}-parallelogram-4` },
  { prop: { star: true }, className: `${selector}-star` },
  { prop: { star2: true }, className: `${selector}-star-2` },
]

const componentProps = [...componentPropsColors, ...componentPropsDefault, ...componentPropsShapes]

const testConfig = {
  testName,
  componentName,
  component,
  componentProps,
  defaultProps,
  selector,
  skipTests,
  refType,
} as TestGeneratorProps

testGenerator(testConfig)
