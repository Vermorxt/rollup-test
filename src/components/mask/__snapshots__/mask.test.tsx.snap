// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@cms/core/[object Object] should create snapshot 1`] = `
<div>
  <img
    alt=""
    class="
          mask  
          
           
        "
    src="/static/media/png/placeholder.png"
  />
</div>
`;
